IssueTracker/DataAccess/Utils/CheckRole.cs

```diff
@@ -0,0 +1,36 @@
+﻿using DataAccess.Data.IData;
+
+namespace DataAccess.Utils
+{
+    public class CheckRole
+    {
+
+        public async static Task<bool> IsOwner(IParticipantData _participantData,Guid userid, int projectId)
+        {
+            var results =await _participantData.GetByProjectIdAsync("spParticipant_GetOwner", projectId);
+            if(results == null) return false;
+            if (results.Count() == 0) return false;
+            var result = results.FirstOrDefault(x => x.UserId == userid);
+            if(result == null) return false;
+            return true;
+
+        }
+        public async static Task<bool> OwnerExists(IParticipantData _participantData, int projectId)
+        {
+            var results = await _participantData.GetByProjectIdAsync("spParticipant_GetOwner", projectId);
+            if (results.Count() == 0) return false;
+            return true;
+        }
+        public async static Task<bool> IsOwnerOrColllab(IParticipantData _participantData, Guid userid, int projectId)
+        {
+            var results = await _participantData.GetByProjectIdAsync(
+                "spParticipant_GetOwnersAndCollabsByProjectId", projectId);
+            if (results == null) return false;
+            if (results.Count() == 0) return false;
+            var result = results.FirstOrDefault(x => x.UserId == userid);
+            if (result == null) return false;
+            return true;
+
```

Comment:
Maybe it is simpler and elegant to omit the first 2 "ifs" and use only the last one for all the checking. Using the null check on the results:
var result = results?.FirstOrDefault(....

Reply:
done